### PR TITLE
added expect_test_id & indexing for click_by_text

### PR DESCRIPTION
**Pull request by petrmanak**

### CODE REVIEW

File: libs/langchain/langchain/tools/playwright/click_by_text.py [hunk 1]
```diff
@@ -33,10 +34,16 @@ class ClickByTextTool(BaseBrowserTool):
     playwright_timeout: float = 3_000
```
Comment (jakub-profiq) [inline libs/langchain/langchain/tools/playwright/click_by_text.py]: You can remove this timeout, as we don't use it now.

Comment (petrmanak) [reply libs/langchain/langchain/tools/playwright/click_by_text.py]: done

File: libs/langchain/langchain/tools/playwright/click_by_text.py [hunk 2]
```diff
@@ -92,16 +99,18 @@ async def _arun(
...
+                playwright_cmd = f"await page.getByRole('{selector_effective}').getByText('{text}').click({{strict:{str(self.playwright_strict).lower()}, timeout:{timeout}}});\n"
                 await awrite_to_file(msg=f'    {playwright_cmd}')
             else:
                 # if not visible, try to click on element only by text
-                await page.get_by_text(text).click(timeout=self.playwright_timeout)
+                text_effective = self._selector_effective(selector=text, index=index)
+                await page.get_by_text(text).click(timeout=timeout)
                 # write playwright command to temp file
-                playwright_cmd = f"await page.getByText('{text}').click({{timeout: {self.playwright_timeout}}});\n"
+                playwright_cmd = f"await page.getByText('{text_effective}').click({{strict:{str(self.playwright_strict).lower()}, timeout:{timeout}}});\n"
```
Comment (jakub-profiq) [inline libs/langchain/langchain/tools/playwright/click_by_text.py]: You cannot to use this syntax. Please try to run this generated code directly i playwright. It is not working. 

Correct typescript playwright command looks like these examples:
await page.getByText('Authentication flow').nth(0).click({timeout:3000});
await page.getByRole('button').getByText('Next').nth(0).click({timeout:3000});

Generated commands looks like this:
await page.getByRole('button >> visible=1 >> nth=0').getByText('Login').click({strict:false, timeout:3000});
await page.getByText('Authentication flow >> visible=1 >> nth=0').click({strict:false, timeout:3000});

**Review by jakub-profiq — CHANGES_REQUESTED** [review]

File: libs/langchain/langchain/tools/playwright/click_by_text.py (hunk 2 above)
Comment (petrmanak) [reply libs/langchain/langchain/tools/playwright/click_by_text.py]: You are absolutely right. The generated code was nonsense, so I rewrote it & test it and it works now.

Comment (jakub-profiq) [reply libs/langchain/langchain/tools/playwright/click_by_text.py]: Amazing, now it looks great. Thank you!

**Pull request merged**